ariadne/cardano/src/Ariadne/Wallet/Backend/Util.hs

```diff
@@ -0,0 +1,103 @@
+module Ariadne.Wallet.Backend.Util
+       ( allAccountIds
+       , ourAccountIdFromAccounts
+       , accountsToUse
+       ) where
+
+import Control.Lens (ix)
+import qualified Data.Text.Buildable
+import Formatting (bprint, build, int, (%))
+
+import Pos.Util (maybeThrow)
+
+import Ariadne.Wallet.Backend.KeyStorage (resolveWalletRefThenRead)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet
+  (HdAccount, HdAccountId, HdRootId, hdAccountId, hdAccountIdParent)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Read (readAccountsByRootId)
+import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet, (@+))
+import Ariadne.Wallet.Cardano.WalletLayer (PassiveWalletLayer(..))
+import Ariadne.Wallet.Face
+  (LocalAccountReference(..), WalletReference, WalletSelection(WSAccount))
+
+
+data OurAccountException = OurAccountNoAccounts !HdRootId
+                         | OurAccountIndexOutOfRange !HdRootId !Word
+                       deriving (Show)
+
+instance Buildable OurAccountException where
+    build = \case
+        OurAccountNoAccounts walletIdx ->
+            bprint ("Wallet "%build%" does not have any accounts") walletIdx
+        OurAccountIndexOutOfRange rootId idx ->
+            bprint ("Account #"%int%" doesn't exist in "%build) idx rootId
+
+instance Exception OurAccountException where
+    displayException = toString . prettyL
+
+-- | Used for fee estimate
```

Comment:
Since it's used _only_ for fee estimation, I'd inline it to the place where it's used (because it's actually small), but I don't insist.